frontend/src/modules/scaffold/views/scaffoldAssetView.js

```diff
@@ -1,342 +1,307 @@
+  var BackboneForms = require('backbone-forms');
+  var Origin = require('core/origin');
+  var AssetManagementModalView = require('modules/assetManagement/views/assetManagementModalView');
+  var AssetCollection = require('modules/assetManagement/collections/assetCollection');
+  var ContentCollection = require('core/collections/contentCollection');
+  var CourseAssetModel = require('core/models/courseAssetModel');
+
+  var ScaffoldAssetView = Backbone.Form.editors.Base.extend({
+    tagName: 'div',
+    events: {
+      // triggered whenever something happens that affects the result of `this.getValue()`
+      'change input': function() {
+        this.toggleFieldAvailibility();
+        this.trigger('change', this);
+      },
+      // triggered whenever an input in this editor becomes the `document.activeElement`
+      'focus input': function() {
+        this.trigger('focus', this);
+      },
+      // triggered whenever an input in this editor stops being the `document.activeElement`
+      'blur input': function() {
+        this.trigger('blur', this);
+      },
+      'click .scaffold-asset-picker': 'onAssetButtonClicked',
+      'click .scaffold-asset-external': 'onExternalAssetButtonClicked',
+      'click .scaffold-asset-clear': 'onClearButtonClicked',
+      'click .scaffold-asset-external-input-save': 'onExternalAssetSaveClicked',
+      'click .scaffold-asset-external-input-cancel': 'onExternalAssetCancelClicked',
+      'click .scaffold-asset-clear-external': 'onExternalClearButtonClicked'
+    },
+
+    initialize: function(options) {
+      this.listenTo(Origin, 'scaffold:assets:autofill', this.onAutofill);
+      Backbone.Form.editors.Base.prototype.initialize.call(this, options);
+    },
+
+    onAutofill: function(courseAssetObject, value) {
+      this.value = value;
+      this.createCourseAsset(courseAssetObject);
+    },
+
+    render: function() {
+      var template = Handlebars.templates[this.constructor.template];
+      var templateData = {
+        value: this.value,
+        type: this.schema.fieldType.replace('Asset:', '')
+      };
+      // this delgate function is async, so does a re-render once the data is loaded
+      var _renderDelegate = _.bind(function(assetId) {
+        if(assetId) {
+          templateData.url = '/api/asset/serve/' + assetId;
+          templateData.thumbUrl = '/api/asset/thumb/' + assetId;
+          this.$el.html(template(templateData));
+        }
+      }, this);
+      // don't have asset ID, so query courseassets for matching URL && content ID
+      this.fetchCourseAsset({
+        _fieldName: this.value.split('/').pop(),
+        _contentTypeId: Origin.scaffold.getCurrentModel().get('_id')
+      }, function(error, collection) {
+        if(error) {
+          console.error(error);
+          return _renderDelegate();
+        }
+        if(collection.length === 0) {
+          console.warn('No asset data found for', Origin.scaffold.getCurrentModel().get('_id'));
```

Comment:
This warning fires a lot; maybe every time there is an asset field without one selected.